lib/metrics.js

```diff
@@ -23,82 +21,95 @@ const prometheus_metrics = [
-  let received = {dyno:formation_type};
-  let alamo_app_name = common.alamo.app_name(app_name, formation_type);
-  let prometheus_url = config.prometheus_metrics_url
-  if(!prometheus_url) {
-    prometheus_url = await common.alamo.region_api_by_space(pg_pool, space_name)
+async function query_prometheus_metrics(
+  pg_pool,
+  fqdn,
+  app_name,
+  space_name,
+  from,
+  to,
+  resolution,
+  formation_type,
+  /* callback */
+) {
+  const received = { dyno: formation_type };
+  const alamo_app_name = common.alamo.app_name(app_name, formation_type);
+  let prometheus_url = config.prometheus_metrics_url;
+  if (!prometheus_url) {
+    prometheus_url = await common.alamo.region_api_by_space(pg_pool, space_name);
   }
-  await Promise.all(prometheus_metrics.map(async function(metric) {
-    let q = metric.indexOf('cpu_') > -1 ? 
-          query_rate_template : 
-          ( metric.indexOf('network_') > -1 ? 
-            query_named_pod_template : 
-            query_template );
-    let query_url = prometheus_url + 
-      '/api/v1/query_range?query=' + 
-      encodeURIComponent(q.replace(/%metric%/, metric).replace(/%space%/, space_name).replace(/%app%/, alamo_app_name).replace(/%resolution%/, resolution)) + 
-      '&start=' + (from.getTime()/1000) + 
-      '&end=' + (to.getTime()/1000) +
-      '&step=' + resolution;
-    let data = await httph.request('get', query_url, {}, null);
-    let result = JSON.parse(data).data.result;
-    let name = metric.replace(/container\_/, '');
+  await Promise.all(prometheus_metrics.map(async (metric) => {
+    const q = metric.indexOf('cpu_') > -1
+      ? query_rate_template
+      : (metric.indexOf('network_') > -1
+        ? query_named_pod_template
+        : query_template);
+    const query_url = `${prometheus_url
+    }/api/v1/query_range?query=${
+      encodeURIComponent(q.replace(/%metric%/, metric).replace(/%space%/, space_name).replace(/%app%/, alamo_app_name).replace(/%resolution%/, resolution))
+    }&start=${from.getTime() / 1000
+    }&end=${to.getTime() / 1000
+    }&step=${resolution}`;
+    const data = await httph.request('get', query_url, {}, null);
+    const { result } = JSON.parse(data).data;
+    const name = metric.replace(/container_/, '');
     received[name] = {};
-    result.forEach((x) => x.values.forEach((y) => received[name][Math.floor(y[0])] = y[1]));
+    result.forEach((x) => x.values.forEach((y) => {
+      [, received[name][Math.floor(y[0])]] = y;
+    }));
   }));
   return received;
 }
 
-async function query_available_influxdb_metrics(pg_pool, fqdn, app_name, space_name, formation_type, callback) {
-  let influx_url = config.influxdb_metrics_url
-  if(!influx_url) {
-    influx_url = await common.alamo.region_api_by_space(pg_pool, space_name)
+async function query_available_influxdb_metrics(pg_pool, fqdn, app_name, space_name, formation_type /* callback */) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Async function 'query_available_influxdb_metrics' has a complexity of 9. Maximum allowed is 8.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)